From 4fb2096c3a19fbe3846956728b2d91aa0ce02135 Mon Sep 17 00:00:00 2001
From: Benjamin Drung <benjamin.drung@profitbricks.com>
Date: Tue, 30 May 2017 11:11:39 +0200
Subject: [PATCH] Install sysvinit script on Debian/Ubuntu

On Debian/Ubuntu systems, install the sysvinit script (in addition to
the systemd service file).
---
 src/CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -84,6 +84,8 @@
     set(CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} ${FASTNETMON_PROFILE_FLAGS}")
 endif()
 
+execute_process(COMMAND sh -c ". /etc/os-release; echo $ID" OUTPUT_VARIABLE OS_ID ERROR_QUIET)
+
 ### Executables definition 
 
 # Main tool
@@ -607,6 +609,10 @@
 configure_file(fastnetmon.service.in "${CMAKE_CURRENT_BINARY_DIR}/fastnetmon.service" @ONLY)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/fastnetmon.service" DESTINATION ${CMAKE_INSTALL_SYSTEMD_SERVICEDIR})
 
+if (${OS_ID} MATCHES debian|ubuntu)
+install(FILES fastnetmon_init_script_debian_6_7 DESTINATION ${CMAKE_INSTALL_SYSCONFDIR}/init.d RENAME fastnetmon)
+endif()
+
 # Configure cpack package builder
 # Run it with: cd build; cpack -G DEB ..
 set(CPACK_PACKAGE_NAME "fastnetmon")
